src/core/tests/type_prop/assign.cpp

```diff
@@ -78,3 +79,99 @@ TEST(type_prop, variable_comparison) {
     ASSERT_FALSE(variable1->get_info() == variable4->get_info());
     ASSERT_FALSE(variable1->get_info() == variable5->get_info());
 }
+
+TEST(type_prop, assign_v6_static_shape_match) {
+    auto input = make_shared<ov::op::v0::Parameter>(element::f32, Shape{1, 2, 64, 64});
+    auto variable = std::make_shared<op::util::Variable>(
+        op::util::VariableInfo{PartialShape{1, 2, 64, 64}, element::f32, "variable_id"});
+    std::shared_ptr<ov::op::v6::Assign> assign;
+    EXPECT_NO_THROW(assign = std::make_shared<ov::op::v6::Assign>(input, variable));
```

Comment:
```suggestion
    auto assign = std::make_shared<ov::op::v6::Assign>(input, variable);
```
I think there is no much benefit of this assertion as there is assumption the the operator should be created with success if there will be error test fill fail anyway.